query/README.md

```diff
@@ -1,66 +1,203 @@
+available that can vastly improve your editor experience when working with MLIR,
+namely [`clangd` for C++ files](https://clangd.llvm.org) and [the MLIR-specific
+language servers for `.mlir`, `.pdll`, and `.td` files](https://mlir.llvm.org/docs/Tools/MLIRLSP).
+Below are instructions for setting them up with a few different editors.
+
+### Visual Studio Code
+
+If you're on macOS, you can install VS Code via Homebrew:
 
 ```sh
-git clone https://github.com/llvm/llvm-project.git
-mkdir llvm-project/build
-cd llvm-project/build
-git checkout llvmorg-15.0.3  # latest stable LLVM/MLIR release
-
-# Some notes:
-#     1. -G Ninja generates a build.ninja file rather than makefiles it's not
-#        required but is recommended by LLVM
-#     2. The CMAKE_INSTALL_PREFIX I put here is a subdirectory of the mlir-hail
-#        (this repo's) root. If you do this, add that directory to
-#        .git/info/exclude and it will be like adding it to a gitignore
-#     3. On linux, using lld via -DLLVM_ENABLE_LLD=ON can speed up the build due
-#        to faster linking.
-#
-# The -DLLVM_BUILD_EXAMPLES=ON flag is optional.
-cmake ../llvm -G Ninja \
-   -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;mlir" \
-   -DLLVM_BUILD_EXAMPLES=ON \
-   -DLLVM_TARGETS_TO_BUILD="AArch64;X86;NVPTX;AMDGPU" \
-   -DCMAKE_BUILD_TYPE=Release \
-   -DLLVM_ENABLE_ASSERTIONS=ON \
-   -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ \
-   -DCMAKE_INSTALL_PREFIX=$HAIL_DIR/query/.dist/llvm \
-   -DCMAKE_EXPORT_COMPILE_COMMANDS=1
-ninja # this will take a while
-ninja install
+brew install --cask visual-studio-code
 ```
 
-## Building the hail-query-mlir project
+Install the [clangd](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd)
+and [MLIR](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-mlir)
+extensions, either by clicking the `Install` button on their webpages, or by
+navigating to the `Extensions` icon in the bar on the left-hand side of the editor
+window, searching for them by name, and clicking the `Install` button.
+
+Edit VS Code's [`settings.json`](https://code.visualstudio.com/docs/getstarted/settings#_settingsjson)
+file and add the following before the `}` at the end of the file:
 
-To set up the build, from this directory:
+```json
+    "mlir.pdll_server_path": "$LLVM_BUILD_BIN_DIRECTORY/mlir-pdll-lsp-server",
+    "mlir.server_path": "$LLVM_BUILD_BIN_DIRECTORY/mlir-lsp-server",
+    "mlir.tablegen_server_path": "$LLVM_BUILD_BIN_DIRECTORY/tblgen-lsp-server",
+    "mlir.tablegen_compilation_databases": [
+        "$HAIL_NATIVE_COMPILER_BUILD_DIRECTORY/tablegen_compile_commands.yml"
+    ]
+```
+
+Replace `$LLVM_BUILD_BIN_DIRECTORY` and `$HAIL_NATIVE_COMPILER_BUILD_DIRECTORY`
+with their values, which can be determined by running `echo $LLVM_BUILD_BIN_DIRECTORY`
+and `echo $HAIL_NATIVE_COMPILER_BUILD_DIRECTORY` from your shell.
```

Comment:
Hmm. It might be easier to provide the user with a command instead. This seems to work. We assume developers have `jq` installed.

```
cd $HOME/Library/Application\ Support/Code/User/
cp settings.json settings.json.bak
jq '.["mlir.pdll_server_path"] = "'$LLVM_BUILD_BIN_DIRECTORY'/mlir-pdll-lsp-server"
    .["mlir.server_path"] = "'$LLVM_BUILD_BIN_DIRECTORY'/mlir-lsp-server"
    .["mlir.tablegen_server_path"] = "'$LLVM_BUILD_BIN_DIRECTORY'/tblgen-lsp-server"
    .["mlir.tablegen_compilation_databases"] = [
        "'$HAIL_NATIVE_COMPILER_BUILD_DIRECTORY'/tablegen_compile_commands.yml"
    ]
' settings.json.bak > settings.json
```

```
(base) ➜  hail git:(reduce-memory-requests) cat settings.json
{
  "a": "sfds"
}
(base) ➜  hail git:(reduce-memory-requests) cp settings.json settings.json.bak
jq '. + {
    "mlir.pdll_server_path": "'$LLVM_BUILD_BIN_DIRECTORY'/mlir-pdll-lsp-server",
    "mlir.server_path": "'$LLVM_BUILD_BIN_DIRECTORY'/mlir-lsp-server",
    "mlir.tablegen_server_path": "'$LLVM_BUILD_BIN_DIRECTORY'/tblgen-lsp-server",
    "mlir.tablegen_compilation_databases": [
        "'$HAIL_NATIVE_COMPILER_BUILD_DIRECTORY'/tablegen_compile_commands.yml"
    ]
}' settings.json.bak > settings.json

(base) ➜  hail git:(reduce-memory-requests) cat settings.json 
{
  "a": "sfds",
  "mlir.pdll_server_path": "/Users/dking/projects/llvm-project/build/bin/mlir-pdll-lsp-server",
  "mlir.server_path": "/Users/dking/projects/llvm-project/build/bin/mlir-lsp-server",
  "mlir.tablegen_server_path": "/Users/dking/projects/llvm-project/build/bin/tblgen-lsp-server",
  "mlir.tablegen_compilation_databases": [
    "/Users/dking/projects/hail/query/build/tablegen_compile_commands.yml"
  ]
}
(base) ➜  hail git:(reduce-memory-requests) cat settings.json.bak 
{
  "a": "sfds"
}
```

Reply:
Thanks for the command! Updated the README to use this instead, and to supply a `sed` command for the Neovim configuration.